File: VB/4PosBackOffice.NET/LenB.vb
Module modLenB
    Public Function LenB(ByVal ObjStr As String) As Integer
        If Len(ObjStr) = 0 Then Return 0
        Return System.Text.Encoding.Unicode.GetByteCount(ObjStr)
    End Function

    Public Function LenB(ByVal Obj As Object) As Integer
        If Obj Is Nothing Then Return 0
        Try
            Return Len(Obj)
        Catch
            Try
                Return System.Runtime.InteropServices.Marshal.SizeOf(Obj)
            Catch
                Return -1
            End Try
        End Try
    End Function
End Module

File: C-Sharp/LenB.cs
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Windows.Forms;
using DpSdkEngLib;
using DPSDKOPSLib;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
namespace _4PosBackOffice.NET
{
	static class modLenB
	{
		public static int LenB(string ObjStr)
		{
			if (Strings.Len(ObjStr) == 0)
				return 0;
			return System.Text.Encoding.Unicode.GetByteCount(ObjStr);
		}

		public static int LenB(object Obj)
		{
			if (Obj == null)
				return 0;
			try {
				return Strings.Len(Obj);
			} catch {
				try {
					return System.Runtime.InteropServices.Marshal.SizeOf(Obj);
				} catch {
					return -1;
				}
			}
		}
	}
}
